File: 48-rotate-image/rotate-image.py
class Solution(object):
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        n=len(matrix)
        if n==1:
            return matrix
        for i in range(n):
            for j in range(i,n):
                    matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
        for i in range(len(matrix)):
            matrix[i].reverse()
        return matrix
                

        
        

File: 48-rotate-image/rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        //first the transposé
        int temp=0;
        System.out.println(Arrays.deepToString(matrix));
        for(int i=0;i<matrix.length;i++){
            for(int j=i;j<matrix[0].length;j++){
                temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        //reverse each aray
        for(int k=0;k<matrix.length;k++){
            reverseArray(matrix[k]);

        }
        System.out.println(Arrays.deepToString(matrix));
    }
    public static void reverseArray(int[] array){
        int temp=0;
        int left=0;
        int right=array.length-1;
        while(left<right){
            temp=array[left];
            array[left]=array[right];
            array[right]=temp;
            left++;
            right--;
        }
    }
}